worksheet.rb

```diff
@@ -23,12 +37,124 @@
 # into this data structure, such as "DR0004"
 # and "3rd Feb 2016" and "RD0022"
 
+# data structure blueprint
+# [
+#   {
+#       driver_id: str,
+#       date: [int, int, int],
+#       cost: int,
+#       rider_id: str,
+#       rating: int
+#   }
+# ]
+
+rides_data = [
+    ['DRIVER_ID','DATE','COST','RIDER_ID','RATING'],
+    ['DR0004','3rd Feb 2016','5','RD0022','5'],
+    ['DR0001','3rd Feb 2016','10','RD0003','3'],
+    ['DR0002','3rd Feb 2016','25','RD0073','5'],
+    ['DR0001','3rd Feb 2016','30','RD0015','4'],
+    ['DR0003','4th Feb 2016','5','RD0066','5'],
+    ['DR0004','4th Feb 2016','10','RD0022','4'],
+    ['DR0002','4th Feb 2016','15','RD0013','1'],
+    ['DR0003','5th Feb 2016','50','RD0003','2'],
+    ['DR0002','5th Feb 2016','35','RD0066','3'],
+    ['DR0004','5th Feb 2016','20','RD0073','5'],
+    ['DR0001','5th Feb 2016','45','RD0003','2']
+]
```

Comment:
You have a 2D array here, also called an array of arrays.  This works, but you have to use the 1st row for the titles of each field in the subsequent sub-arrays.

You could instead have an array of hashes.

```ruby
rides_data = [
  {
    driver_id: "DR0004",
    date: "3rd Feb 2016",
    cost: 5,
    rider_id: "RD0022",
    rating: 5,
  },
  ...
]
```